### PR TITLE
Resource and Data Source Implementation for DFP Service

### PR DESCRIPTION
**Pull request by unasra**

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -73,6 +74,7 @@ func (p *BloxOneProvider) Configure(ctx context.Context, req provider.ConfigureR
 		option.WithClientName(fmt.Sprintf("terraform/%s#%s", p.version, p.commit)),
 		option.WithAPIKey(data.APIKey.ValueString()),
 		option.WithCSPUrl(data.CSPUrl.ValueString()),
+		option.WithDebug(true),
```
Comment (mathewab) [inline internal/provider/provider.go:77]: debug should be false by default

File: docs/resources/dfp_service.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  display_name = "example_host"
+}
+resource "bloxone_infra_service" "example" {
+  name           = "example_dfp_service"
+  pool_id        = bloxone_infra_host.example_host.pool_id
+  service_type   = "dfp"
+  desired_state  = "start"
+  wait_for_state = false
+
+}
```
Comment (mathewab) [inline docs/resources/dfp_service.md:26]: use a data source instead. 

```suggestion
data "bloxone_infra_services" "example" {
  name = "example_dfp_service"
}
```

File: internal/service/dfp/api_dfp_resource.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, _, err := r.client.DNSForwardingProxyAPI.
+		InfraServicesAPI.
+		CreateOrUpdateDfpService(ctx, utils.ExtractResourceId(data.ServiceId.ValueString())).
```
Comment (mathewab) [inline internal/service/dfp/api_dfp_resource.go:71]: `ExtractResourceId` should be handled in go-client like we do for all other resources.

File: internal/provider/provider.go (hunk 1 above)
Comment (unasra) [reply internal/provider/provider.go]: Fixed

File: internal/service/dfp/api_dfp_resource.go (hunk 3 above)
Comment (unasra) [reply internal/service/dfp/api_dfp_resource.go]: Fixed

File: internal/provider/provider.go [hunk 4]
```diff
@@ -129,6 +128,7 @@ func (p *BloxOneProvider) Resources(_ context.Context) []func() resource.Resourc
 
 		dfp.NewDfpResource,
 
+		fw.NewSecurityPoliciesResource,
 		fw.NewAccessCodesResource,
 		fw.NewNamedListsResource,
 		fw.NewNetworkListsResource,
```
Comment (mathewab) [inline internal/provider/provider.go]: nit: not for this pr, but all of these resource names here should be "singular".

**Review by mathewab — APPROVED** [review]

File: internal/provider/provider.go (hunk 4 above)
Comment (unasra) [reply internal/provider/provider.go]: Will fix in a mis PR with other changes

**Pull request merged**